plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/BaseBigQueryConnectorTest.java

```diff
@@ -659,6 +689,42 @@ public void testMissingWildcardTable()
                 .hasMessageEndingWith("does not match any table.");
     }
 
+    @Test(dataProvider = "testColumnNameFailsDataProvider")
+    public void testColumnNameFails(String columnName)
+    {
+        Throwable throwable = null;
+        try {
+            super.testColumnName(columnName);
+            fail("Test above should throw");
+        }
+        catch (Exception e) {
+            throwable = e;
+        }
+        catch (Error e) {
+            throwable = e.getCause();
+        }
+        assertThat(throwable.getMessage()).containsPattern(String.join("|",
+                ".*Failed to convert table schema to Avro record schema. field name %s, error Invalid name: %s..*".formatted(columnName, columnName),
+                ".*Invalid field name \"%s\". Fields must contain the allowed characters, and be at most 300 characters long..*".formatted(columnName.replace("\\", "\\\\")),
+                ".*Illegal initial character: %s.*".formatted(columnName)
+        ));
+    }
+
+    @DataProvider
+    public Object[][] testColumnNameFailsDataProvider()
+    {
+        return NOT_ALLOWED_ANYMORE_COLUMN_NAMES.stream().collect(toDataProvider());
+    }
+
+    @Override
+    protected Optional<String> filterColumnNameTestData(String columnName)
+    {
+        if (NOT_ALLOWED_ANYMORE_COLUMN_NAMES.contains(columnName)) {
```

Comment:
Creating and inserting into tables with these columns works and selecting the table isn't problem, right? This change hides the fact and we can't understand what happened after the operations. It would be better to deny such columns in all operations, or verify the result with different way, e.g. use query based API instead of storage API.